File: app/src/main/java/com/memoizlabs/kotlin_playground/currency/CurrencyGson.java
package com.memoizlabs.kotlin_playground.currency;

public class CurrencyGson {

    private Rates rates;

    public GBPExchangeRate toGbpExchangeRate() {
        return new GBPExchangeRate(new Money(rates.USD), new Money(rates.EUR));
    }

    @Override
    public String toString() {
        return "CurrencyGson{" +
                ", rates=" + rates +
                '}';
    }

    private static class Rates {

        double USD;
        double EUR;

        @Override
        public String toString() {
            return "Rates{" +
                    "USD=" + USD +
                    ", EUR=" + EUR +
                    '}';
        }
    }
}

File: app/src/main/java/com/memoizlabs/kotlin_playground/currency/CurrencyGson.kt
package com.memoizlabs.kotlin_playground.currency

class CurrencyGson {

    private val rates: Rates? = null

    fun toGbpExchangeRate(): GBPExchangeRate {
        return GBPExchangeRate(Money(rates!!.USD), Money(rates.EUR))
    }

    override fun toString(): String {
        return "CurrencyGson{" +
                ", rates=" + rates +
                '}'
    }

    private class Rates {

        internal var USD: Double = 0.toDouble()
        internal var EUR: Double = 0.toDouble()

        override fun toString(): String {
            return "Rates{" +
                    "USD=" + USD +
                    ", EUR=" + EUR +
                    '}'
        }
    }
}
